### ISSUE TITLE
New command type `:named-anonymous` 

### ISSUE THREAD

**Issue by aadcg** [atlas-engineer/nyxt#2808]
@aartaka alluded to this idea [in a comment in the context of prompt buffer actions](https://github.com/atlas-engineer/nyxt/pull/2801#issuecomment-1439966049).

Notice that there's a comment in the codebase that already hints at this idea.

```log
;; TODO: Can we use `alex:named-lambda'?  How do we get the name then?
```

**Comment by Ambrevar** [comment]
Isn't this `lambda-command`?

**Comment by aadcg** [comment]
I don't think so. `lambda-command`s are set locally, but you can't refer to them elsewhere. And binding those to a variable isn't a good solution since the symbol will be bound in the variables namespace, not the functions namespace.

**Comment by Ambrevar** [comment]
Can you give a detailed recipe of the use case?

**Comment by aadcg** [comment]
@Ambrevar, [please read this comment by @aartaka](https://github.com/atlas-engineer/nyxt/pull/2801#issuecomment-1439966049).

**Comment by aadcg** [comment]
> So if we accept functions as actions, the problem is solved I believe.

Yes, that's the conclusion for the time being! 

> Alternatively, we could play with the export status of the command symbols: don't list the command in execute-command if it's not exported, for instance.

And this is another conclusion for the future. Should the need for such commands appear, then we can explore this idea.
